### ISSUE TITLE
Document how to write automated detector tests

### ISSUE THREAD

**Issue by ProfJanetDavis** [gender-bias/gender-bias#69]
There should be documentation of how to test a detector. Maybe this should go in or with the ["how to write a detector" document](https://github.com/gender-bias/gender-bias/tree/master/docs/hacking). @j6k4m8 should weigh in on where this documentation belongs.

**Comment by j6k4m8** [comment]
Agreed! In the tradition of [test-driven development](https://www.agilealliance.org/glossary/tdd/) (which @neiljp expertly enforced in the early days of this repo, much to our benefit!) it probably makes sense for writing tests to be the _backbone_ of that tutorial document. I'll add this to my stack :)

**Comment by j6k4m8** [comment]
Might also make sense for this doc to live in the wiki so that we can update it without a full PR/review process.

**Comment by ProfJanetDavis** [comment]
@j6k4m8, we are on the same page about test-driven development! 

Thanks for putting this on your stack. I'm taking the liberty of assigning this issue to you so that I remember.

**Comment by ProfJanetDavis** [comment]
And I like the idea of putting this documentation in the wiki so that it is easier to keep up to date. I did that with [another project](https://github.com/glam-lab/degender-the-web/wiki) and I think it works pretty well. The only reason I can think of to keep the developer docs in the code base would be if it helps remind contributors to update the docs when they update the code.

**Comment by ProfJanetDavis** [comment]
Hmmm, that makes me think we should go a step beyond test-driven development to do [doc-driven development](https://opensource.com/article/17/8/doc-driven-development) - a [time-honored tradition](https://dl.acm.org/doi/abs/10.1145/30401.30402) in my field of human-computer interaction.
